### PR TITLE
new: android rework

### PR DESCRIPTION
**Pull request by SunriseFox**
+ remove unused popup.html and clear index.html (It's confusing that we have to have an index.html)
+ splice browser-polyfill from firefox
+ add browser.permissions polyfill for geckoview
+ copy polyfills even when dev rebuild

### CODE REVIEW

File: src/polyfill/permissions.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+;(()=>{
+  if (!browser) return
+  if (browser.permissions) return
+  browser.permissions = {}
+  const requestedUrls = []
```
Comment (Jack-Works) [inline src/polyfill/permissions.js]: You should save them to the localstorage or somewhere.

**Comment by SunriseFox** [comment]
I think it is ok to not use set on such a small dataset, a set will add some extra overhead.
This script is only loaded from options-page, whenever the user opens it. So maybe not reloading on getAll is also ok.

**Comment by Jack-Works** [comment]
> This script is only loaded from options-page, whenever the user opens it. So maybe not reloading on getAll is also ok.

When generate backups, it is called in the background script.

**Comment by SunriseFox** [comment]
So... If `getAll` is called by background page, it means... no matter how I change this file here, it never affects the result being retrieved from the background page. Background page uses native ways to get and request permissions and this patch only applies to wherever there's not a `browser.permissions`.

Shall it replace the native `browser.permissions` anyway?

**Comment by Jack-Works** [comment]
If you're patching it on the options page, it should also be patched in the background page

**Comment by SunriseFox** [comment]
Should different artifact generated by different branches? I think you should settle it.

**Comment by Jack-Works** [comment]
> Should different artifact generated by different branches? I think you should settle it.

Yes that will be nice. I'll do it afternoon.

**Pull request closed without merging**